cluster-essentials/clusterinit.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package clusteressentials has APIs for bootstrapping kubernetes cluster
+package clusteressentials
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/client-go/rest"
+
+	clusterutil "github.com/vmware-tanzu/tanzu-framework/cluster-essentials/utilinterface"
+)
+
+type ClusterEssentialsConfiguration struct {
+	ClusterEssentialBundlePath string
+	RestConfig                 *rest.Config
+	PkgClient                  clusterutil.Client
+}
+
+const tmpDir = "tmp"
+
+func (c *ClusterEssentialsConfiguration) InitClusterEssentials() error {
+	err := c.PkgClient.DownloadClusterEssentialBundle(c.ClusterEssentialBundlePath)
+	if err != nil {
+		return fmt.Errorf("unable to download cluster essential manifest: %w", err)
+	}
+
+	packages, err := os.ReadDir(tmpDir)
+	if err != nil {
+		return err
+	}
+	for _, item := range packages {
+		packageDir := filepath.Join(tmpDir, item.Name())
+		generatedManifestBytes, err := c.PkgClient.CarvelPackageProcessor(packageDir)
+		if err != nil {
+			return fmt.Errorf("unable to process carvel package: %w", err)
+		}
+		err = c.PkgClient.ApplyResourcesFromManifest(generatedManifestBytes, c.RestConfig)
```

Comment:
Let's say there are two packages in the bundle & the second one is corrupted. This results in the cluster having resources belonging to first package installed resulting in a partially installed state. Instead, we can start creating resources only if all the packages are processed successfully.